java-generator/core/src/main/java/io/fabric8/java/generator/CRGeneratorRunner.java

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.java.generator;
+
+import com.github.javaparser.ast.CompilationUnit;
+import io.fabric8.java.generator.nodes.AbstractJSONSchema2Pojo;
+import io.fabric8.java.generator.nodes.GeneratorResult;
+import io.fabric8.java.generator.nodes.JCRObject;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinition;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinitionSpec;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinitionVersion;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import java.io.File;
+import java.util.*;
+
+public class CRGeneratorRunner {
+
+    public void run(File source, File basePath) {
+        try (final KubernetesClient client = new DefaultKubernetesClient()) {
+            // Parse CRD with fabric8
+            CustomResourceDefinition crd =
+                    client.apiextensions().v1().customResourceDefinitions().load(source).get();
+
+            List<WritableCRCompilationUnit> writables =
+                    generate(crd, getPackage(crd.getSpec().getGroup()));
+
+            for (WritableCRCompilationUnit w : writables) {
+                w.writeAllJavaClasses(basePath);
+            }
+        }
+    }
+
+    public List<WritableCRCompilationUnit> generate(
+            CustomResourceDefinition crd, Optional<String> basePackageName) {
+        CustomResourceDefinitionSpec crSpec = crd.getSpec();
+        String crName = crSpec.getNames().getKind();
+        String group = crSpec.getGroup();
+
+        List<WritableCRCompilationUnit> writableCUs =
+                new ArrayList<WritableCRCompilationUnit>(crSpec.getVersions().size());
+        for (CustomResourceDefinitionVersion crdv : crSpec.getVersions()) {
+            CompilationUnit cu = new CompilationUnit();
+
+            String version = crdv.getName();
+
+            String pkg = basePackageName.map((p) -> p + "." + version).orElse(version);
+
+            cu.setPackageDeclaration(pkg);
+
+            AbstractJSONSchema2Pojo specGenerator = null;
+
+            if (crdv.getSchema().getOpenAPIV3Schema().getProperties().get("spec") != null) {
```

Comment:
Why are you not using local variables?

Reply:
my religion .... 😛  just joking, you are right I was simply going along with complexity